scripts/libraries/solvers/MulticallDexes.ts

```diff
@@ -117,11 +120,22 @@ export default class MulticallDexes implements Solver {
       //   });
     }
 
+    // allowance checks
+    for (const response of uniqueSwapResponse ? [uniqueSwapResponse] : dexesBestResults) {
+      const token = await IERC20Metadata__factory.connect(response.path[0], tradeFactory.signer);
+      const { swapperAddress, router, amountOut, dex } = response;
+      const approveToken = (await token.allowance(swapperAddress, router)).lt(amountOut);
```

Comment:
```suggestion
      const approveToken = (await token.allowance(multicallAddress, router)).lt(amountOut);
```
The check on allowance should not be from the swapper since "Swapper" might be the "UniswapSwapper" address. The allowance check is from multicall.